server/src/main/java/org/elasticsearch/index/shard/IndexShard.java

```diff
@@ -779,10 +779,26 @@ public void relocated(
         final String targetAllocationId,
         final BiConsumer<ReplicationTracker.PrimaryContext, ActionListener<Void>> consumer,
         final ActionListener<Void> listener
+    ) throws IllegalIndexShardStateException, IllegalStateException {
+        relocated(targetNodeId, targetAllocationId, consumer, listener, null);
+    }
+
+    /**
+     * Provides an variant of {@link IndexShard#relocated(String, String, BiConsumer, ActionListener, Releasable)} with an option
+     * to relocate the shard under externally acquired primary permits.
+     *
+     * @param acquiredPrimaryPermits if null, waits until all the primary permits are acquired, otherwise completes relocation immediately
```

Comment:
I think that this comment is incorrect, it should be `otherwise it calls the consumer immediately`, right?

Reply:
I've updted the comment